src/averageAges.js

```diff
@@ -37,7 +39,15 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWomen = people.filter(person =>
+    withChildren ? people.some(children =>
+      children.mother === person.name)
+      && person.sex === 'f' : (person.sex === 'f'));
+
+  const womenAge = filteredWomen.map(person => person.died - person.born);
+  const totalWomenAge = womenAge.reduce((sum, age) => sum + age);
```

Comment:
You do the same on `calculateMenAverageAge`. You can create a separate function and use it here and in `calculateMenAverageAge`.